src/middleware.api.ts

```diff
@@ -45,14 +45,51 @@ export const middleware = async (request: NextRequest) => {
     return NextResponse.redirect(url);
   }
 
-  const isOwnershipPage = request.nextUrl.pathname.endsWith('ownerships');
+  const isOwnershipPage =
+    (request.nextUrl.pathname.startsWith('/organizer') &&
+      !request.nextUrl.pathname.endsWith('/ownerships')) ||
+    request.nextUrl.pathname.startsWith('/search');
 
-  if (isOwnershipPage && !process.env.NEXT_PUBLIC_OWNERSHIP_ENABLED) {
-    const url = new URL('/404', request.url);
-    return NextResponse.redirect(url);
+  if (isOwnershipPage) {
+    const isOwnershipEnabled =
+      process.env.NEXT_PUBLIC_OWNERSHIP_ENABLED === 'true';
```

Comment:
I would maybe bypass this entire condition if this is not the case, I'd even argue leaving the extra search param in is harmless. Would allow you to really simplify the logic and merge conditions I think

Reply:
I'm going to merge it for now, so other PRs can use this logic